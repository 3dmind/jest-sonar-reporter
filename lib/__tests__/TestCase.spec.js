'use strict'

const xml = require('xml')
const TestCase = require('../TestCase')

describe('TestCase', () => {
  test('<testCase name="" duration=""/>', () => {
    const mock = {
      title: 'lorem ipsum'
    }
    const testCase = new TestCase(mock)

    const actualReport = xml(testCase)

    expect(actualReport).toMatchSnapshot()
  })

  test('failing test case', () => {
    const mock = {
      title: 'lorem ipsum',
      status: 'failed',
      failureMessages: ['Lorem ipsum']
    }
    const testCase = new TestCase(mock)

    const actualReport = xml(testCase, true)

    expect(actualReport).toMatchSnapshot()
  })
})
