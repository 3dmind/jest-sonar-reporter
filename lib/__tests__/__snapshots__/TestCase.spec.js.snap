// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestCase <testCase name="" duration=""/> 1`] = `"<testCase name=\\"lorem ipsum\\" duration=\\"0\\"/>"`;

exports[`TestCase failing test case 1`] = `
"<testCase name=\\"lorem ipsum\\" duration=\\"0\\">
    <failure message=\\"Lorem ipsum\\"/>
</testCase>"
`;
